File: pl-domain/department-service/src/main/java/com/polylab/department/app/mapper/DepartmentDtoMapper.java
package com.polylab.department.app.mapper;

import com.jxareas.jxcore.common.mapper.TwoWayMapper;
import com.polylab.department.app.dto.DepartmentDto;
import com.polylab.department.persistence.model.Department;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.modelmapper.ModelMapper;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class DepartmentDtoMapper implements TwoWayMapper<DepartmentDto, Department> {
    private final ModelMapper mapper;

    @Override
    public Department mapTo(@NotNull DepartmentDto source) {
        return mapper.map(source, Department.class);
    }

    @Override
    public DepartmentDto mapFrom(@NotNull Department destination) {
        return mapper.map(destination, DepartmentDto.class);
    }
}

File: pl-domain/department-service/src/main/java/com/polylab/department/app/mapper/DepartmentDtoMapper.kt
package com.polylab.department.app.mapper

import com.jxareas.efficax.annotations.Mapper
import com.jxareas.efficax.domain.mapper.MirrorMapper
import com.jxareas.efficax.extensions.mirrorMap
import com.polylab.department.app.dto.DepartmentDto
import com.polylab.department.persistence.model.Department

@Mapper
object DepartmentDtoMapper : MirrorMapper<DepartmentDto, Department> by mirrorMap()
